src/main/java/bio/terra/cli/service/utils/WorkspaceManagerService.java

```diff
@@ -705,4 +709,41 @@ public void deleteControlledBigQueryDataset(UUID workspaceId, UUID resourceId) {
           "Error deleting controlled Big Query dataset in the workspace.", ex);
     }
   }
+
+  /**
+   * Helper method that checks a JobReport's status and throws an exception if it's not COMPLETED.
+   *
+   * <p>- Throws a {@link SystemException} if the job FAILED.
+   *
+   * <p>- Throws a {@link UserActionableException} if the job is still RUNNING. Some actions are
+   * expected to take a long time (e.g. deleting a bucket with lots of objects), and a timeout is
+   * not necessarily a failure. The action the user can take is to wait a bit longer and then check
+   * back (e.g. by listing the buckets in the workspace) later to see if the job completed.
+   *
```

Comment:
Not for this PR, but do we have any future planned stories around asynchronous commands that allow for completion polling directly through CLI (as opposed to observing side effects)?

Reply:
We've talked about it briefly a couple times so far, and always decided to punt. I just added a [ticket](https://broadworkbench.atlassian.net/browse/PF-745) to make sure we don't forget about it.